prediction_market_agent/tools/prediction_prophet/research.py

```diff
@@ -0,0 +1,39 @@
+from prediction_market_agent_tooling.loggers import logger
+from prediction_prophet.functions.research import research
+from pydantic.types import SecretStr
+
+
+def prophet_research(
+    goal: str,
+    model: str,
+    openai_api_key: SecretStr,
+    tavily_api_key: SecretStr,
+    initial_subqueries_limit: int = 20,
+    subqueries_limit: int = 4,
+    max_results_per_search: int = 5,
+    min_scraped_sites: int = 10,
+) -> str:
+    """
+    Use `min_scraped_sites` as a proxy for setting a minimum requirement for
+    'how thorough the research must be'. Up to (min_scraped_sites * max_results_per_search)
+    sites will be scraped, but the actual number may be less because of:
+
+    - duplication of URLs across searches
+    - failed scrapes
+    - insufficient search results (although unlikely)
+
+    If the number of scraped sites is less than `min_scraped_sites`, an error
+    will be raised.
+    """
+    return research(
+        goal=goal,
+        model=model,
+        use_summaries=False,
+        initial_subqueries_limit=initial_subqueries_limit,
+        subqueries_limit=subqueries_limit,
+        max_results_per_search=max_results_per_search,
+        min_scraped_sites=min_scraped_sites,
+        openai_api_key=openai_api_key,
+        tavily_api_key=tavily_api_key,
+        logger=logger,
```

Comment:
use our own logger!

Reply:
Note that our logger is used to show messages in Strealit apps: https://github.com/gnosis/prediction-market-agent/blob/f55a4648748602372630ba383c6e1708a5d11568/prediction_market_agent/tools/streamlit_utils.py#L49

So if this is changed, all the messages logged in the prophet_research will be shown as well, which I don't mind if they are or aren't, just an side-effect to have in mind.

Reply:
Ah good point - I hadn't considered that! Had a look with `streamlit run scripts/agent_app.py` and I prefer it with the extra logging - and it made me see that we don't use our logger for the research calls in `DeployablePredictionProphet*Agent`s (because they call `research` indirectly, and that call is inside the predictionprophet library), which makes for a bit of an empty response in the app
<img width="691" alt="Screenshot 2024-08-08 at 19 22 59" src="https://github.com/user-attachments/assets/ddb0dd81-33b5-4fd9-81a6-54a3c35562bc">

Made this issue to track https://github.com/gnosis/prediction-market-agent/issues/391 (can't do it in this PR now because it requires a change to predictionprophet).